File: src/main/java/com/example/dpop/orchestrator/dpop/DpopProof.java
package com.example.dpop.orchestrator.dpop;

import com.nimbusds.jose.jwk.JWK;

import java.time.Instant;

public record DpopProof(
        String token,
        JWK publicKey,
        String jti,
        String htm,
        String htu,
        Instant issuedAt,
        String nonce
) {
}

File: src/main/kotlin/com/example/dpop/orchestrator/dpop/DpopProof.kt
package com.example.dpop.orchestrator.dpop

import com.nimbusds.jose.jwk.JWK
import java.time.Instant

@JvmRecord
data class DpopProof(
    val token: String,
    val publicKey: JWK,
    val jti: String,
    val htm: String,
    val htu: String,
    val issuedAt: Instant,
    val nonce: String?
)
